nautobot_golden_config/views.py

```diff
@@ -410,7 +410,11 @@ def diff_structured_data(backup_data, intended_data):
 
             settings = get_device_to_settings_map(queryset=Device.objects.filter(pk=device.pk))
             if device.id in settings:
-                _, output = graph_ql_query(request, device, settings[device.id].sot_agg_query.query)
+                sot_agg_query_setting = settings[device.id].sot_agg_query
+                if sot_agg_query_setting is not None:
+                    _, output = graph_ql_query(request, device, sot_agg_query_setting.query)
+                else:
+                    output = None
```

Comment:
This should only cover settings that have been created before the validationerror above was added? Making sure I understand how this would ever be hit

Reply:
That is my understanding as well.

Reply:
@jeffkala the validation error is only triggered when saving the settings. If SOTAGG_ENABLED is toggled back and forth, then it is possible to have SOTAGG_ENABLED=False and configure the settings without a GraphQL query. If the SOTAGG_ENABLED is later set to True, then the settings would not be re-validated, and the code that triggers #393 would still be triggered.

I don't think it is likely a common scenario, and generally having this check will solve the problem, and help the users know what needs to be configured based on the plugin config.